homeassistant/components/sabnzbd/sensor.py

```diff
@@ -14,33 +15,34 @@
 from homeassistant.const import UnitOfDataRate, UnitOfInformation
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
-from homeassistant.helpers.dispatcher import async_dispatcher_connect
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
 
-from . import DOMAIN, SIGNAL_SABNZBD_UPDATED
-from .const import DEFAULT_NAME, KEY_API_DATA
+from . import DOMAIN, SabnzbdUpdateCoordinator
+from .const import DEFAULT_NAME
 
 
 @dataclass(frozen=True, kw_only=True)
 class SabnzbdSensorEntityDescription(SensorEntityDescription):
     """Describes Sabnzbd sensor entity."""
 
     key: str
+    value_fn: Callable[[StateType], StateType] = lambda value: value
 
 
-SPEED_KEY = "kbpersec"
-
 SENSOR_TYPES: tuple[SabnzbdSensorEntityDescription, ...] = (
     SabnzbdSensorEntityDescription(
         key="status",
         translation_key="status",
     ),
     SabnzbdSensorEntityDescription(
-        key=SPEED_KEY,
+        key="kbpersec",
         translation_key="speed",
         device_class=SensorDeviceClass.DATA_RATE,
         native_unit_of_measurement=UnitOfDataRate.MEGABYTES_PER_SECOND,
         state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda value: float(value) / 1024 if value is not None else None,
```

Comment:
Can we in the future set the native_unit_of_measurement to the native unit and then set the suggested UoM to MB/s?

Reply:
Oh I see you are adding this now, maybe that can be a preliminary PR to avoid the whole value_fn field to be added in the first place

Reply:
Makes sense, have changed that.